index_datas/import_datas.py

```diff
@@ -0,0 +1,130 @@
+
```

Comment:
**[Desired,Maintainability]** This code is pretty messy and can be cleaned up a lot. Below is a cleaner version, It will still need to be tested though:
```python
import gzip
from lxml import etree
from elasticsearch import Elasticsearch as es
from multiprocessing import Pool

elastic_search = es(["127.0.0.1"], timeout=35,
                    max_retries=8, retry_on_timeout=True)

"""
Need to create the index first before using this script
curl -XPUT https://localhost:9200/example3 --insecure -u admin:admin
"""
URL = 'https://localhost:9200/example3/_doc/'

def index_elastic_search(data):
    global elasticsearch
    try:
        elastic_search.index(index='news', body=data)
        print("Added document: ", data["title"])
    except (Exception) as e:
        print("Could not index document, {0}:".format(data["title"]), e)


def search_index_test(elastic_search):
    # u6250082 Xuguang Song
    '''test with query to match ACT'''

    q = "ACT"
    position = "title"
    test1 = {
        "query": {
            "match": {
                position: q
            }
        }
    }
    # output = elastic_search.search(index='news', doc_type='web_news', body=test1)
    print('\n', 'searching for keyword: ', q, " ", "in", " article ", position, '\n')
    output = elastic_search.search(index='news', body=test1)
    print('searching finished with total time: ', output['took'], '\n')
    print('result: ', '\n')
    for hit in output['hits']['hits']:
        # print search result
        print('match news title: ', hit['_source']['title'], '\n', 'match news link: ', hit['_source']['link'], '\n', '-------------------------------------------------------------')

def generate_abstract_elements():
    with gzip.open('../../wikidump/enwiki-20210820-abstract.xml.gz', 'rb') as f:
        for _, element in etree.iterparse(f, events=('end',), tag='doc'):
            title = element.findtext('./title')
            link = element.findtext('./url')
            art = element.findtext('./abstract')
            yield {"link": link, "title": title, "art": art}
            element.clear()

def main():
    global doc_id, lock
    if not elastic_search.indices.exists(index='news'):
        elastic_search.indices.create(index='news')
    totalLines = 0
    pool = Pool(10)
    result_iter = pool.imap_unordered(index_elastic_search, generate_abstract_elements(), chunksize=50)
    for _ in result_iter:
        totalLines += 1

    print('In Total:', totalLines)
    search_index_test(elastic_search)

if __name__ == '__main__':
    main()
```

Reply:
don't want to risk too much but I reorganized the codes

Reply:
Risk? What risk? I'm simply cleaning up the code, risk is eliminated when you test this.